core-tests/shared/src/test/scala/zio/concurrent/ReentrantLockSpec.scala

```diff
@@ -0,0 +1,110 @@
+package zio.concurrent
+
+import zio.duration._
+import zio.test.Assertion._
+import zio.test.TestAspect.{flaky, timeout}
+import zio.test._
+import zio.{Exit, Promise, Ref, Schedule, ZIO}
+
+object ReentrantLockSpec extends DefaultRunnableSpec {
```

Comment:
This is copied from `TReentrantLockSpec`. As long as `ReentrantLock` is just delegating to `TReentrantLock` there is no real benefit in this spec. However, it might be useful if `ReentrantLock` is changed to another underlying encoding in the future.